pkg/reconciler/ingress/ingress.go

```diff
@@ -224,28 +225,34 @@ func (r *Reconciler) reconcileVirtualServices(ctx context.Context, ing *v1alpha1
 	}
 
 	// Now, remove the extra ones.
-	vses, err := r.virtualServiceLister.VirtualServices(ing.GetNamespace()).List(
-		labels.SelectorFromSet(labels.Set{networking.IngressLabelKey: ing.GetName()}))
-	if err != nil {
-		return fmt.Errorf("failed to get VirtualServices: %w", err)
+	selectors := map[string]string{
+		networking.IngressLabelKey: ing.GetName(), // VS created since 0.12
+		serving.RouteLabelKey:      ing.GetName(), // VS created before 0.12
 	}
+	for k, v := range selectors {
+		vses, err := r.virtualServiceLister.VirtualServices(ing.GetNamespace()).List(
+			labels.SelectorFromSet(labels.Set{k: v}))
+		if err != nil {
+			return fmt.Errorf("failed to get VirtualServices: %w", err)
+		}
 
-	// Sort the virtual services by their name to get a stable deletion order.
-	sort.Slice(vses, func(i, j int) bool {
-		return vses[i].Name < vses[j].Name
-	})
+		// Sort the virtual services by their name to get a stable deletion order.
```

Comment:
Does this matter here?

Reply:
I haven't changed any of this code, it is just inside a loop now